pkg/operators/oci-handler/oci_test.go

```diff
@@ -0,0 +1,356 @@
+
+			result, err := getPullSecret(test.secretName, test.secretNamespace, fcs)
+
+			if test.expectError {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), test.expectedErrorMsg)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, test.expectedResult, string(result))
+			}
+		})
+	}
+}
+
+func TestConstructTempConfig(t *testing.T) {
+	tests := []struct {
+		name       string
+		input      string
+		wantErr    bool
+		wantConfig map[string]any
+		isField    bool
+	}{
+		{
+			name:    "Invalid no colon",
+			input:   "invalidannotation",
+			wantErr: true,
+		},
+		{
+			name:    "Invalid no equal sign",
+			input:   "source:invalidannotation",
+			wantErr: true,
+		},
+		{
+			name:    "Valid datasource annotation",
+			input:   "source:foo=bar",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"annotations": map[string]any{
+							"foo": "bar",
+						},
+					},
+				},
+			},
+		},
+		{
+			name:    "Valid field annotation",
+			input:   "source.field:foo=bar",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"fields": map[string]any{
+							"field": map[string]any{
+								"annotations": map[string]any{
+									"foo": "bar",
+								},
+							},
+						},
+					},
+				},
+			},
+			isField: true,
+		},
+		{
+			name:    "Invalid multiple colons",
+			input:   "source:field:foo=bar",
+			wantErr: true,
+		},
+		{
+			name:    "multiple equal signs (base64 encoded)",
+			input:   "source:foo=bar=baz",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"annotations": map[string]any{
+							"foo": "bar=baz",
+						},
+					},
+				},
+			},
+		},
+		{
+			name:    "multiple colon signs in value (base64 encoded)",
+			input:   "source:foo=b:a:r",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"annotations": map[string]any{
+							"foo": "b:a:r",
+						},
+					},
+				},
+			},
+		},
+		{
+			name:    "multiple dots (for subfields)",
+			input:   "source.field.foo:foo=bar",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"fields": map[string]any{
+							"field.foo": map[string]any{
+								"annotations": map[string]any{
+									"foo": "bar",
+								},
+							},
+						},
+					},
+				},
+			},
+			isField: true,
+		},
+		{
+			name:    "Empty annotation key",
+			input:   "source:=bar",
+			wantErr: true,
+		},
+		{
+			name:    "Empty annotation value",
+			input:   "source:foo=",
+			wantErr: true,
+		},
+		{
+			name:    "Empty subject",
+			input:   ":foo=bar",
+			wantErr: true,
+		},
+		{
+			name:    "Empty field name",
+			input:   "source.:foo=bar",
+			wantErr: true,
+		},
+		{
```

Comment:
```suggestion
		{
			name:    "Empty subfield name",
			input:   "source.test..bar:foo=bar",
			wantErr: true,
		},
		{
```

I added a new testcase, which should fail, but is accepted by the proposed code changes in the first commit